src/cc/frontends/clang/b_frontend_action.h

```diff
@@ -174,7 +174,7 @@ class BFrontendAction : public clang::ASTFrontendAction {
   void DoMiscWorkAround();
   // negative fake_fd to be different from real fd in bpf_pseudo_fd.
   int get_next_fake_fd() { return next_fake_fd_--; }
-  void add_map_def(int fd, std::tuple<int, std::string, int, int, int, int> map_def) {
+  void add_map_def(int fd, std::tuple<int, std::string, int, int, int, int, std::string> map_def) {
     fake_fd_map_[fd] = map_def;
```

Comment:
`move` here as well?

Reply:
Hi, do you mean `fake_fd_map_[fd] = move(map_def);`?

Reply:
General approach looks okay. Could you change the above last element in map_def from `std::string` to `int`? In any case, it is expected to be an int to represent the pinned map id.

Reply:
@pothos Yeah, just a nit, since we already the tuple constructed:)

Reply:
Yeah, just a nit, since we already constructed the tuple~

Reply:
Ok, done.